### ISSUE TITLE
Possible misuse of aria-label; Div element with tabindex=0

### ISSUE THREAD

**Issue by rvantonisse** [validator/validator#1349]
Posting an issue here with instructions from the validator;

> Warning: Possible misuse of aria-label. (If you disagree with this warning, [file an issue report](https://github.com/validator/validator/issues/new) or send e-mail to www-validator@w3.org.)

Code validated:

```html
<div id="uuid_ab264e74-29cd-4bfc-bfc2-959b7eb476d6_1" data-label="New slideshow" data-loop='1'  aria-label='Some descriptive text' tabindex='0'>
```

I think the issue here is that the `div` is missing a role rather than misusing the `aria-label`. The element is made keyboard accessible / focusable with `tabindex=0` and thus requires to have a name ( and role). This makes the use of aria-label valid here. (also if the full text content does not make sense, or is too descriptive, a shorter label / name may be more helpful)

The contents of `aria-label` may be wrong here, but that depends on the role, also required.

**Comment by scottaohara** [comment]
Due to the div missing a role, it's a misuse of aria-label.  ARIA 1.2 prohibits the use of aria-label on generic elements, regardless of them being focusable or not.  This is related to my issue https://github.com/validator/validator/issues/1334

**Comment by sideshowbarker** [comment]
Thanks for taking time to report this. https://github.com/validator/validator/issues/1349#issuecomment-1081895246 seems to resolve this, so I’m going ahead and closing it.

**Comment by rvantonisse** [comment]
Ok, so to wrap up:

Giving a name with `aria-label` or `aria-labelledby` to an html element without a specific role will be a misuse when ARIA 1.2 reaches "recomendation" status. Untill then this is an "experimental" validator warning.

[1349#issuecomment-1081895246](https://github.com/validator/validator/issues/1349#issuecomment-1081895246):
> Due to the div missing a role, it's a misuse of aria-label. ARIA 1.2 prohibits the use of aria-label on generic elements, regardless of them being focusable or not.

ARIA 1.2 "Roles which cannot be named":
1. [ARIA 1.2 "Accessible Name Calculation"](https://www.w3.org/TR/wai-aria-1.2/#namecalculation) 
2. [ARIA 1.2 "Roles which cannot be named (Name prohibited)"](https://www.w3.org/TR/wai-aria-1.2/#namefromprohibited)

[Nu Html Checker](https://validator.w3.org/nu/):
> This tool is an ongoing experiment in better HTML checking, and its behavior remains subject to change

**Comment by stevefaulkner** [comment]
@rvantonisse to put a bow on your wrap up

> Giving a name with aria-label or aria-labelledby to an html element without a specific role will be a misuse when ARIA 1.2 reaches "recommendation" status. Until then this is an "experimental" validator warning.

Whether or not a specification has met a particular milestone in the W3C standards process is not the best point of consideration. The HTML checker is not designed to _validate_ code as good or bad, it is designed to provide indications to developers of coding practices that as implemented can have a negative effect upon, amongst other things, user experience. A developer is not going to sent to the _naughty chair_ for not following the advice.  It is known through testing with browsers and assistive technology that on certain elements `aria-label/aria-labelledby` is not well supported or not supported at all. The checker surfaces this as a service to developers who wish to improve the quality and interoperability of their code. You as an individual can choose to follow the advice or not.

This article [HTML5 – Check it Before you Wreck it with Mike[tm] Smith](https://html5doctor.com/html5-check-it-before-you-wreck-it-with-miketm-smith/) is from 2014, but contains @sideshowbarker's (still relevant in 2022) thoughts at the time on his ongoing development and maintenance of the HTML checker:

> So I call the tool at [validator.w3.org/nu](http://validator.w3.org/nu) the Nu Markup Checker instead of the blah-blah-Validator because I want to spread the happiness of the word checker in the sense of doing something actually useful for people instead of just giving them a pat on the back. It’s an automated thing which checks stuff for you that’d otherwise be really tedious for you to check manually. So it helps you. Maybe it should be called the Nu Help-You Checker.
